bugbug/tools/code_review_generation.py

```diff
@@ -0,0 +1,754 @@
+
+            if not deleted_lines or not added_lines:
+                logger.error(f"No deleted or added lines found for file: {filename}")
+                return None
+
+            deletion_start_diff_line = deleted_lines[
+                min(
+                    len(deleted_lines) - 1,
+                    max(0, start_line - deletion_start_line - hunk_size),
+                )
+            ]
+            deletion_end_diff_line = deleted_lines[
+                max(
+                    0,
+                    min(
+                        len(deleted_lines) - 1,
+                        end_line - deletion_start_line + hunk_size,
+                    ),
+                )
+            ]
+
+            addition_start_diff_line = added_lines[
+                min(
+                    len(added_lines) - 1,
+                    max(0, start_line - addition_start_line - hunk_size),
+                )
+            ]
+            addition_end_diff_line = added_lines[
+                max(
+                    0,
+                    min(
+                        len(added_lines) - 1, end_line - addition_start_line + hunk_size
+                    ),
+                )
+            ]
+
+            first_index = None
+            last_index = None
+
+            diff_lines = match.group(0).split("\n")
+
+            for i, line in enumerate(diff_lines):
+                if line in [
+                    deletion_start_diff_line,
+                    deletion_end_diff_line,
+                    addition_start_diff_line,
+                    addition_end_diff_line,
+                ]:
+                    if first_index is None:
+                        first_index = i
+                    last_index = i
+
+        if first_index is not None and last_index is not None:
+            relevant_diff = "\n".join(diff_lines[first_index : last_index + 1])
+            return relevant_diff
+        else:
+            logger.error(f"No relevant diff found for lines: {start_line}-{end_line}")
+            return None
+    else:
+        logger.error(f"No diff found for file: {filename}")
+        return None
+
+
+def get_revision_id_from_patch(patch_id):
+    diffs = api.search_diffs(diff_id=patch_id)
+
+    if diffs:
+        revision_phid = diffs[0]["revisionPHID"]
+
+        revision = api.load_revision(rev_phid=revision_phid)
+
+        return revision["id"]
+    else:
+        logger.error(f"No diffs found for patch ID: {patch_id}")
+        return None
+
+
+def fetch_diff(revision_id, patch_id):
+    try:
+        url = f"https://phabricator.services.mozilla.com/D{revision_id}?id={patch_id}&download=true"
+        response = requests.get(url)
+        response.raise_for_status()
+        return response.text
+    except requests.HTTPError as e:
+        logger.error(f"HTTP error fetching diff: {e}")
+        return None
+    except Exception as e:
+        logger.error(f"Unexpected error: {e}")
+        return None
```

Comment:
For the functions that retrieve data from Phab, we should move them to the phabricator.py lib (and use already existing ones in case there are)